src/net_processing.cpp

```diff
@@ -2758,6 +2757,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         if (nCMPCTBLOCKVersion != CMPCTBLOCKS_VERSION) return;
 
         LOCK(cs_main);
+        // Only support compact block relay with witness peers
+        if (!State(pfrom.GetId())->fHaveWitness) return;
```

Comment:
According to BIP144, `NODE_WITNESS` means 

> A node will signal that it can provide witnesses using the following service bit

So I'm not sure this is strictly necessary. As a general theme though, I'm not sure bundling this logic with the value of NODE_WITNESS is necessary/correct.

Reply:
Agree. I've now removed it.